sklearn/metrics/ranking.py

```diff
@@ -171,16 +191,48 @@ def average_precision_score(y_true, y_score, average="macro",
     >>> y_true = np.array([0, 0, 1, 1])
     >>> y_scores = np.array([0.1, 0.4, 0.35, 0.8])
     >>> average_precision_score(y_true, y_scores)  # doctest: +ELLIPSIS
-    0.79...
+    0.83...
+
+    >>> yt = np.array([0, 0, 1, 1])
+    >>> ys = np.array([0.1, 0.4, 0.35, 0.8])
+    >>> ap = average_precision_score(yt, ys, interpolation='eleven_point')
+    >>> ap # doctest: +ELLIPSIS
+    0.84...
 
     """
-    def _binary_average_precision(y_true, y_score, sample_weight=None):
+    def _binary_uninterpolated_average_precision(
+            y_true, y_score, sample_weight=None):
         precision, recall, thresholds = precision_recall_curve(
             y_true, y_score, sample_weight=sample_weight)
-        return auc(recall, precision)
+        # Return the step function integral
+        return -np.sum(np.diff(recall) * np.array(precision)[:-1])
```

Comment:
I think this works because the last entry of precision is guaranteed to be 1, as written in the docstring of `precision_recall_curve`; do you think this warrants a note in a comment here, for posterity?